src/combinatorics/simple_graph/triangle/basic.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.clique
+
+/-!
+# Triangles in graphs
+
+This file defines triangles in simple graphs.
+
+A triangle is a `3`-clique, that is a set of three vertices which are all connected.
+
+## Main declarations
+
+* `simple_graph.triangle_finset`: Finset of triangles of a graph.
+* `simple_graph.triangle_free`: Predicate for a graph to have no triangle.
+* `simple_graph.triangle_free_far`: Predicate for a graph to have enough triangles that one must
+  remove a lot of edges to remove all of them. This is the crux of the Triangle Removal lemma.
+-/
+
+open finset fintype
+
+namespace simple_graph
+variables {α 𝕜 : Type*} [linear_ordered_field 𝕜] (G H : simple_graph α) {n : ℕ} {s : finset α}
+
+section triangle_free
+
+/-- `G.triangle_free` means that `G` has no triangles. -/
+def triangle_free : Prop := ∀ t, ¬ G.is_n_clique 3 t
```

Comment:
I guess you want to generalise this?

Reply:
To cliques? I think the correct generalization is induced subgraphs, but the definition would be a whole different game.

Reply:
To cliques, yeah. I agree the optimal generalisation is induced subgraphs, but cliques would be a reasonable middle ground, and should be pretty easy?

Reply:
Should I also generalize `triangle_finset` to `n_clique_finset`?

Reply:
I would argue so, yes